client/my-sites/plans-features-main/index.tsx

```diff
@@ -476,39 +453,24 @@ const PlansFeaturesMain = ( {
 	} );
 
 	// we neeed only the visible ones for comparison grid (these should extend into plans-ui data store selectors)
-	const gridPlansForComparisonGrid = useMemo( () => {
-		const hiddenPlans = [ PLAN_HOSTING_TRIAL_MONTHLY, PLAN_ENTERPRISE_GRID_WPCOM ];
-
-		return filteredPlansForPlanFeatures.reduce( ( acc, gridPlan ) => {
-			if ( gridPlan.isVisible && ! hiddenPlans.includes( gridPlan.planSlug ) ) {
-				return [
-					...acc,
-					{
-						...gridPlan,
-						features: planFeaturesForComparisonGrid[ gridPlan.planSlug ],
-					},
-				];
-			}
-
-			return acc;
-		}, [] as GridPlan[] );
-	}, [ filteredPlansForPlanFeatures, planFeaturesForComparisonGrid ] );
+	const plansForComparisonGrid = usePlansForComparisonGrid( {
+		allFeaturesList: FEATURES_LIST,
+		gridPlans: filteredPlansForPlanFeatures,
+		intent,
+		selectedFeature,
+		showLegacyStorageFeature,
+	} );
 
 	// we neeed only the visible ones for features grid (these should extend into plans-ui data store selectors)
-	const gridPlansForFeaturesGrid = useMemo( () => {
-		return filteredPlansForPlanFeatures.reduce( ( acc, gridPlan ) => {
-			if ( gridPlan.isVisible ) {
-				return [
-					...acc,
-					{
-						...gridPlan,
-						features: planFeaturesForFeaturesGrid[ gridPlan.planSlug ],
-					},
-				];
-			}
-			return acc;
-		}, [] as GridPlan[] );
-	}, [ filteredPlansForPlanFeatures, planFeaturesForFeaturesGrid ] );
+	const plansForFeaturesGrid = usePlansForFeaturesGrid( {
+		allFeaturesList: FEATURES_LIST,
+		availablePlans: gridPlans || [],
```

Comment:
I chose the prop name `availablePlans` to sit alongside the other `gridPlans` prop, based on references to 'available' in the internal implementation of `useGridPlans`. Appreciate any suggestions or feedback on this naming.

Reply:
This feels a bit weird that we are passing `availablePlans` here (I'd rename to `availableGridPlans` like the others) and `gridPlans` below.

Can you clarify what the difference is and how they are being used?

Reply:
I agree. In the followup, only `gridPlans` will remain. For the time being, `availableGridPlans` is used by the `usePlanFeaturesForGridPlans` call within `useGridPlansForFeaturesGrid`. 

I've updated the naming to `availableGridPlans`: [7d2f3c2](https://github.com/Automattic/wp-calypso/pull/87377/commits/7d2f3c2944a99ddb5c2c4fd8dbc92fafddcc4c2d)